src/NUnitTestExtractor/NUnitTestExtractor/Program.cs

```diff
@@ -55,44 +57,78 @@ static void Main(string[] args)
         /// <param name="outputDirectory"> the directory in that the file is being saved to </param>
         /// <param name="dllDirectories"> list of all dll directories which the user wishes to append to a file or stdout </param>
         /// <param name="level"> Specifies the level of granuality to use for the output tests entered by user. </param>
-        private static void WriteTestDllAndName(string name, string outputDirectory, IList<string> dllDirectories, Level level)
+        private static void WriteTestDllAndName(StreamWriter writer, string dll, string data)
         {
-            //user entered some dlls
-            if (dllDirectories.Count != 0)
+            using (writer)
             {
-                StreamWriter writer;
-
-                if (outputDirectory == null)
+                writer.Write(dll + "|" + data + Environment.NewLine);
+            }           
+        }
+        /// <summary>
+        /// Uses reflection in order to search user submitted dlls to find either namespaces, classes, or functions which contain tests, then uses WriteTestDllAndName() to
+        /// write data to either stdout or a txt file
+        /// </summary>
+        /// <param name="dlls">The dlls which are submitted by user</param>
+        /// <param name="level">Specifies the level of granuality to use for the output: either namepspace, class or function</param>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Reliability", "CA2001:AvoidCallingProblematicMethods", MessageId = "System.Reflection.Assembly.LoadFrom")]
+        private static void GetTests(IList<string> dlls, Level level)
+        {
+            foreach(string dll in dlls)
+            {
+                Assembly assembly = null;
+                try
                 {
-                    writer = new StreamWriter(Console.OpenStandardOutput());
-                }
-                else
+                    assembly = Assembly.LoadFrom(dll);
+                }catch(FileNotFoundException e)
                 {
-                    writer = new StreamWriter(outputDirectory + name +".txt", true);
+                    Console.Error.WriteLine("{0}: file {1} was not found on system", e.GetType(), dll);
+                    continue;
                 }
-                foreach (string dll in dllDirectories)
+               
+                foreach (Type type in assembly.GetTypes())
                 {
-                    if (File.Exists(dll))
-                    {
-                        writer.Write("{0}|{1}", dll, level + Environment.NewLine);
-                    }
-                    else
+                    foreach (MethodInfo methodInfo in type.GetMethods())
                     {
-                        Console.Error.WriteLine("Error: Unable to append {0} as it cannot be found on the system", dll);
+                        var attributes = methodInfo.GetCustomAttributes(true);
+                        foreach (var attr in attributes)
```

Comment:
Add a blank line between different blocks of code (like before a loop or if/switch statement...) to make it easier to read.